app/lib/dartdoc/backend.dart

```diff
@@ -133,10 +133,41 @@ class DartdocBackend {
     return versions.map((pv) => pv.version).take(limit).toList();
   }
 
+  bool _shouldRetryUploadError(e) {
+    if (e is DetailedApiRequestError) {
+      return e.status == 502 || e.status == 503;
+    }
+    return false;
+  }
+
+  Future _uploadWithRetry(
+      String objectName, Stream<List<int>> openStream()) async {
+    for (int attempt = 3; attempt > 0; attempt--) {
```

Comment:
Don't we already have a function for the retry pattern?

Reply:
We have one both in pana and in pub site, slightly different ones. I've updated the code and the utils, and I think it shouldn't be too far out to merge this with the one in pana. It would be beneficial if we had a common utility package for such shared code or contribute them to one of the async package utils.